cosmrs/src/crypto/public_key.rs

```diff
@@ -120,6 +121,24 @@ impl TryFrom<&Any> for PublicKey {
     }
 }
 
+impl TryFrom<cosmos_sdk_proto::cosmos::crypto::ed25519::PubKey> for PublicKey {
+    type Error = ErrorReport;
+
+    fn try_from(public_key: cosmos_sdk_proto::cosmos::crypto::ed25519::PubKey) -> Result<PublicKey> {
+        let tm_key = tendermint::public_key::PublicKey::from_raw_secp256k1(&public_key.key);
+        tm_key.map(Into::into).ok_or_else(|| Error::Crypto.into())
```

Comment:
I think this might be a bit easier to read if you get rid of the intermediate variable:

```suggestion
        tendermint::public_key::PublicKey::from_raw_secp256k1(&public_key.key)
            .map(Into::into)
            .ok_or_else(|| Error::Crypto.into())
```

Same goes for below.

Reply:
Done